internal/controllers/common_controller.go

```diff
@@ -118,3 +118,20 @@ func setDeprecationStatusesUnknown(conditions *[]metav1.Condition, message strin
 		})
 	}
 }
+
+// setInstalledStatusConditionFailed sets the installed status condition to failed.
+func setDeletingStatusConditionFailed(conditions *[]metav1.Condition, message string, generation int64) {
+	apimeta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               ocv1alpha1.TypeDeleting,
+		Status:             metav1.ConditionFalse,
```

Comment:
This one is tricky. I think this combo of `TypeDeleting` and `ConditionFalse` would be interpreted as "this extension is not in the deleting state"

And I suppose that "DeletionFailed" could mean "I'm no longer actively in the deleting state because the deletion attempt failed".

But what happens at this point? Does `kapp-controller` keep retry-ing the deletion?

Reply:
Digging into the codebase (haven't verified yet by trying it out), looks like it doesn't. If the deletion fails, the process is still marked as completed (https://github.com/carvel-dev/kapp-controller/blob/d89101eb78b2c26a73a7afd8e8ae3111a6731dde/cli/pkg/kctrl/cmd/app/app_tailer.go#L52) and the watches are closed (https://github.com/carvel-dev/kapp-controller/blob/d89101eb78b2c26a73a7afd8e8ae3111a6731dde/cli/pkg/kctrl/cmd/app/app_tailer.go#L58). 

Going a bit further and looking into the PackageInstall reconciler to see what happens - looks like if deletion fails, and the App's status is conveys the same - the packageInstall just updates its status: https://github.com/carvel-dev/kapp-controller/blob/d89101eb78b2c26a73a7afd8e8ae3111a6731dde/pkg/packageinstall/packageinstall.go#L408-L428. 

This makes sense, because the reconcile period for the App CR is set by the user (and defaulted to 30sec if it's unset). 

This is why they seem to have the reconcileFailed condition which is set when Deleting is set to false.

Reply:
`kapp-controller` does seem to retry deletion on Apps indefinitely (essentially just a `kapp delete`).

I think you're right about the TypeDeleting being confusing, it reads more like another state of the Install/Reconcile/Progressing set of conditions. A single condition looks a lot cleaner.

Reply:
From what I can tell, `kapp-controller` does seem to keep retrying the App delete, though I'm not sure if there's anything to limit the retries: https://github.com/carvel-dev/kapp-controller/blob/d89101eb78b2c26a73a7afd8e8ae3111a6731dde/pkg/app/app_reconcile.go#L36